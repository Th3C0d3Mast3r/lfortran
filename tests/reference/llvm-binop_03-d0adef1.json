{
    "basename": "llvm-binop_03-d0adef1",
    "cmd": "lfortran --no-color --show-llvm {infile} -o {outfile}",
    "infile": "tests/../integration_tests/binop_03.f90",
    "infile_hash": "bcef994ff506d3b6f05f8864a2dbc53781966babedd977d9735ffa87",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm-binop_03-d0adef1.stdout",
    "stdout_hash": "aa915d1941a3c8721c1090d8c78903e4241eebe1d1c82b23ce921b85",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}