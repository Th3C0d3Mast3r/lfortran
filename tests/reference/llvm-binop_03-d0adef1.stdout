; ModuleID = 'LFortran'
source_filename = "LFortran"

@0 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@serialization_info = private unnamed_addr constant [3 x i8] c"I8\00", align 1
@1 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1
@2 = private unnamed_addr constant [11 x i8] c"ERROR STOP\00", align 1
@3 = private unnamed_addr constant [2 x i8] c"\0A\00", align 1
@4 = private unnamed_addr constant [5 x i8] c"%s%s\00", align 1

define i32 @main(i32 %0, i8** %1) {
.entry:
  %x = alloca i64, align 8
  call void @_lpython_call_initial_functions(i32 %0, i8** %1)
  %x1 = alloca i64, align 8
  store i64 665663010, i64* %x1, align 4
  %2 = load i64, i64* %x1, align 4
  %simplified_pow_operation = mul i64 %2, %2
  store i64 %simplified_pow_operation, i64* %x1, align 4
  %3 = call i8* (i8*, i8*, i32, ...) @_lcompilers_string_format_fortran(i8* null, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @serialization_info, i32 0, i32 0), i32 0, i64* %x1)
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @1, i32 0, i32 0), i8* %3, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @0, i32 0, i32 0))
  %4 = load i64, i64* %x1, align 4
  %5 = icmp ne i64 %4, 443107242882260100
  br i1 %5, label %then, label %else

then:                                             ; preds = %.entry
  call void (i8*, ...) @_lcompilers_print_error(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @4, i32 0, i32 0), i8* getelementptr inbounds ([11 x i8], [11 x i8]* @2, i32 0, i32 0), i8* getelementptr inbounds ([2 x i8], [2 x i8]* @3, i32 0, i32 0))
  call void @exit(i32 1)
  br label %ifcont

else:                                             ; preds = %.entry
  br label %ifcont

ifcont:                                           ; preds = %else, %then
  call void @_lpython_free_argv()
  br label %return

return:                                           ; preds = %ifcont
  ret i32 0
}

declare void @_lpython_call_initial_functions(i32, i8**)

declare i8* @_lcompilers_string_format_fortran(i8*, i8*, i32, ...)

declare void @_lfortran_printf(i8*, ...)

declare void @_lcompilers_print_error(i8*, ...)

declare void @exit(i32)

declare void @_lpython_free_argv()
